src/execution/operator/join/physical_range_join.cpp

```diff
@@ -227,11 +231,18 @@ idx_t PhysicalRangeJoin::LocalSortedTable::MergeNulls(const vector<JoinCondition
 		}
 	}
 
-	auto &primary = keys.data[0];
 	if (all_constant == keys.data.size()) {
 		//	Either all NULL or no NULLs
+		if (ConstantVector::IsNull(primary)) {
```

Comment:
I don't think this is needed. All that was wrong with this branch was the call to `ConstantVector::SetNull(primary, true);`

Reply:
We still need to merge all nulls to `validity` of `primary` even in constant branch, since later `primary` will used as sort key. 
This line is a fast path without copying `ValidityMask`, it is not necessary though.

Reply:
This looks correct.